pkg/state/hcl_loader.go

```diff
@@ -0,0 +1,288 @@
+package state
+
+import (
+	nativejson "encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/imdario/mergo"
+	"github.com/variantdev/dag/pkg/dag"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/json"
+	"go.uber.org/zap"
+
+	"github.com/helmfile/helmfile/pkg/filesystem"
+)
+
+const (
+	badIdentifierDetail         = "A name must start with a letter or underscore and may contain only letters, digits, underscores, and dashes."
+	helmfileVarsBlockIdentifier = "helmfile_vars"
```

Comment:
Sure ! no problem about it. 
I originally set "variables" because semantically to me, it "sounds" different than a "value". A value is static to me and a variable has a dynamic connotation. But "values" indeed does make more sense in helmfile context. I'll update it accordingly